packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -646,22 +646,48 @@ export default createRule<Options, MessageId>({
           .getCallSignaturesOfType(
             getConstrainedTypeAtLocation(services, callback),
           )
-          .map(sig => sig.getReturnType());
-        /* istanbul ignore if */ if (returnTypes.length === 0) {
-          // Not a callable function
-          return;
```

Comment:
minor regression.... which I _think_ hit this line previously, but I haven't validated that.
```ts
declare const lol: any;
[1, 2, 3].filter(lol); // should not flag
```

Reply:
Sweet! Thanks! Yeah, IIRC `any` and `Function` are the two types I know of that are permitted to be called without any TS errors, but don't have any call signatures. Some obscure linting trivia 🤷‍♂️